src/main/java/org/myrobotlab/service/interfaces/InterfaceChangeListener.java

```diff
@@ -0,0 +1,9 @@
+package org.myrobotlab.service.interfaces;
+
+public interface InterfaceChangeListener {
+
+  public void onInterfaceRegistered(String serviceName, String interfaceName);
```

Comment:
this interface should also have a method , where the change listener returns the list of interface types it's interested in listening for.

Reply:
this interface has been removed, because it is no longer applicable